swarm/src/behaviour.rs

```diff
@@ -270,14 +358,10 @@ pub trait NetworkBehaviour: 'static {
     fn inject_dial_failure(
         &mut self,
         peer_id: Option<PeerId>,
-        handler: Self::ConnectionHandler,
         error: &DialError,
+        id: ConnectionId,
```

Comment:
I am assuming that this might simplify a couple of `NetworkBehaviour` implementations. :+1:

Reply:
To be honest, these events are quite a mess, esp. the divergence of `FromSwarm` and `SwarmEvent`. It would be nice if we can get rid of `SwarmEvent` apart from it emitting behaviour events.